app/infrastructure/persistence/repositories/user_repo.py

```
@@ -115,8 +127,20 @@ async def create_user(
             hashed_password=hashed,
             is_active=True,
         )
-        created = await self.create(user)
-        return _user_to_result(created)
+        try:
+            created = await self.create(user)
+            return _user_to_result(created)
+        except IntegrityError:
+            raise UserAlreadyExistsException()
```

Ruff (0.15.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)